modules/@apostrophecms/page/index.js

```diff
@@ -2187,7 +2188,47 @@ database.`);
           }
         }
       },
-      ...require('./lib/legacy-migrations')(self)
+      ...require('./lib/legacy-migrations')(self),
+      addMisreplicatedParkedPagesMigration() {
+        self.apos.migration.add('misreplicated-parked-pages', async () => {
+          const parkedPages = await self.apos.doc.db.find({
+            parkedId: {
+              $exists: 1
+            }
+          }).toArray();
+          let locales = Object.keys(self.apos.i18n.locales);
+          if (locales[0] !== self.apos.i18n.defaultLocale) {
+            locales = [ self.apos.i18n.defaultLocale, ...locales.filter(locale => locale !== self.apos.i18n.defaultLocale) ];
```

Comment:
What you have is fine but believe you could also do something like below if you wanted to avoid filtering since you're just rearranging the array.
```
locales = [ ...new Set([ self.apos.i18n.defaultLocale, ...locales ])
```
Might not even need to do the check if you're setting the order yourself.

Reply:
Right about all of the above, I didn't internalize enough everything set does for me.